packages/react-server/src/ReactFlightServer.js

```diff
@@ -757,6 +764,18 @@ const Dispatcher: DispatcherType = {
   useTransition(): [(callback: () => void) => void, boolean] {
     return [() => {}, false];
   },
+  getCacheForType<T>(resourceType: () => T): T {
+    invariant(
+      currentCache,
+      'Reading the cache is only supported while rendering.',
+    );
+    if (currentCache.has(resourceType)) {
```

Comment:
These look ups need to be super fast because we expect there to be a lot and they get recalled for every rerender.

`has + get` causes two look ups. Instead we should use the pattern:

```
const value = currentCache.get(resourceType);
if (value !== undefined) {
  return value;
}
```

This means that if the resource type ends up actually using the value `undefined`, we end up recalling the initializer. However, that's unusual. Can be considered an anti-pattern. It's also likely that the initializer is fast if the value is undefined.

The exception might be if you implement a fetch that sometimes returns undefined as a value. It'd refetch each time and cause an infinite loop but you'd catch that quick.

Reply:
Yea that's why I did it. I guess we could warn on undefined.